StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/New Payment Method Screen/PaymentMethodTypeCollectionView.swift

```diff
@@ -114,6 +114,8 @@ extension PaymentMethodTypeCollectionView: UICollectionViewDataSource, UICollect
     func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
         collectionView.scrollToItem(at: indexPath, at: .centeredHorizontally, animated: true)
         selected = paymentMethodTypes[indexPath.item]
+        STPAnalyticsClient.sharedClient.logPaymentSheetEvent(event: .paymentSheetCarouselPaymentMethodTapped,
```

Comment:
This view is also being used in customerSheet, so calling it `paymentSheet` may not be accurate.

Reply:
Oh interesting, I'll rename this in that case.

Reply:
Looks like we went with a more generic name.  Seems that we loose some insight into paymentSheet only behavior.  That seems problematic, right?  I think we need to pass in a parameter into this class so that we don't attribute CustomerSheet taps into PaymentSheet taps, and vice versa,  what do you think?

Reply:
The string is fairly generic that we will see in Hubble `"mc_carousel_payment_method_tapped"`. We should be able to differentiate between CustomerSheet and PaymentSheet one of two ways:

1. Looking at `product_usage`, this might not work well if someone is using CustomerSheet and PaymentSheet.
2. Looking at the session ID, and tying other PaymentSheet specific events together

However this does make querying more difficult, I'm going to update this so it only sends when being used with PaymentSheet as that is the true intention, and I'm not sure how this event ties into the CustomerSheet health story. If we want to send this event for CustomerSheet it should have a different name such as `"cs_carousel_payment_method_tapped" `or something.